sql/core/src/main/scala/org/apache/spark/sql/execution/python/WindowInPandasExec.scala

```diff
@@ -80,11 +80,11 @@ import org.apache.spark.util.Utils
  * window.
  */
 case class WindowInPandasExec(
-    windowExpression: Seq[NamedExpression],
-    partitionSpec: Seq[Expression],
-    orderSpec: Seq[SortOrder],
+    override val windowExpression: Seq[NamedExpression],
+    override val partitionSpec: Seq[Expression],
```

Comment:
Is `override val` required here?

Reply:
same question. `case class` means everything is `val` so this is just adding `override`.

Reply:
@HyukjinKwon @cloud-fan  Thanks! I dropped most changes based on this comment https://github.com/apache/spark/pull/27368#discussion_r376925299. So current refactor rules are:

1. If abstract class or trait defined with `def`, then no override needed in derived class.
2. Change `HashJoin` definition to `def` to avoid `override` in derived classes.
3. Remove abstract class constructer part of `EvalPythonExec` `WindowExecBase` to sync with other operator code style.